src/librustdoc/doctest.rs

```diff
@@ -544,13 +546,17 @@ crate fn make_test(
         let returns_result = everything_else.trim_end().ends_with("(())");
         let (main_pre, main_post) = if returns_result {
             (
-                "fn main() { fn _inner() -> Result<(), impl core::fmt::Debug> {",
-                "}\n_inner().unwrap() }",
+                "fn main() { fn _main_() -> Result<(), impl core::fmt::Debug> {\n",
```

Comment:
Is `test_id` only non-empty when calling this from the `InstrumentCoverage` pass? Or are you proposing changing the inner function name for all doctests from now on? I'm not sure if that's a good thing, or if it would be better to limit the effect of the change. Maybe that change should not be limited to only when coverage is enabled.

A couple of suggestions here:

1. Instead of sticking with the literal `str` for this value, how about just making this a call to `format!()` and avoid the need to `replace` a pattern later. This also avoids the possibility that the pattern might actually be in the test itself, likely causing a bug.

2. Since the `returns_result == true` case already implements a wrapped function, and you're doing the same, can I suggest keeping as much of the original implementation and style as before, so without coverage enabled, things are the same?

For example, would something like this work?

```rust
let wrapped_fn_name = if test_id.is_empty() {
    "_inner"
} else {
    "_inner_" + test_id
};
let (main_pre, main_post) = if returns_result {
    format!("fn main() { fn {}() -> Result<(), impl core::fmt::Debug> {", wrapped_fn_name),
    format!("}\n{}().unwrap() }", wrapped_fn_name),
} else {
    format!("fn main() { fn {}() {", wrapped_fn_name),
    format!("}\n{}() }", wrapped_fn_name),
};
```

This still does make one change to the test structure: if `test_id` is empty and the `returns_result` is false. I'm not sure it's worth the effort to avoid creating the wrapper only in that case, so unless it causes big issues in downstream tests (hard to believe it would), you probably don't need that special case handling.

Reply:
> Or are you proposing changing the inner function name for all doctests from now on?

Yes, as `test_id` is always defined with my proposed change. I think its an easy change to make, and I don’t think it should have any other effects, but I could be wrong.